### DISCUSSION TITLE
Looking for a CFW without Popup when connecting to Wifi

### DISCUSSION THREAD

**Discussion by Rqbln** [arendst/Tasmota#19618] [Q&A]
Good morning,
We are working on a device that connects to an outlet. Upon connection, the device's firmware, based on Tasmota 13.0, initiates a Wi-Fi Access Point (AP) mode and displays a captive portal popup. Our goal is to modify the firmware in such a way that the device operates as a Wi-Fi client without broadcasting a Wi-Fi network or displaying the popup.

Technical Details:
Tasmota Version: 13.0.0
Objective: Remove the captive portal popup when connecting to the Tasmota outlet.
Proposed Method: Modify the Tasmota firmware files to disable the AP mode and prevent the popup from appearing, while still enabling the device to function as a Wi-Fi client.
Files of Interest: We've identified multiple files that might contain logic related to AP mode and the popup. We modified in `my_user_config.h` the `WIFI_CONFIG_TOOL` in all the possible ways, but the pop up only disappear when adding a SSID/password, so we tried modifying other files. These include `support_wifi.ino, support_command.ino, xdrv_01_9_webserver.ino, xdrv_10_scripter.ino, xdrv_57_9_tasmesh.ino, and xdrv_58_range_extender.ino`.

We're also curious to know if anyone in the community has previously embarked on a similar project aimed at removing the captive portal popup in Tasmota. If there are existing contributions or projects, we'd be keen on integrating or collaborating with them.

Feel free to share your ideas, suggestions, or contributions to achieve this objective. Your expertise and help are much appreciated!

**Comment by sfromis** [comment]
If you build Tasmota to already have Wifi credentials baked in, Tasmota will boot without AP mode being used. No code modifications needed.

```
#define STA_SSID1              ""                // [Ssid1] Wi-Fi SSID
#define STA_PASS1              ""                // [Password1] Wi-Fi password
#define STA_SSID2              ""                // [Ssid2] Optional alternate AP Wi-Fi SSID
#define STA_PASS2              ""                // [Password2] Optional alternate AP Wi-Fi password
```

Additionally, if you are doing wired flashing (typically needed for plugs), you can config Wifi credentials using the same serial connection, to avoid using AP mode to select Wifi network.

**Reply by Rqbln** [reply]
Mmmhh our goal is to create firmware that can be deployed to multiple homes without relying on predefined Wi-Fi credentials. We want Tasmota firmware to be able to adapt to different Wi-Fi networks without requiring specific SSID and password settings. We have tested this possibility and it is the only one where we do not get a pop up, but it cannot be of use to us.

**Reply by barbudor** [reply]
There's something I don't get in your logic
If you remove the portal that allows to configurea Wifi network and refuse to hardcode wifi, how are you expecting to provision the device on an unknown network?

**Reply by Rqbln** [reply]
By connecting to the ip of the plug when connected to his wifi, that's the same page as the pop up.
I just don't want it to display when you're connecting to the wifi

**Reply by sfromis** [reply]
The plug cannot become a client until you've configured WiFi credentials for the network it is to be a client on. But now it sounds like you do not want it to be a client, but still an AP just without the captive portal page for onboarding. This is not a Tasmota feature, but you may be interested in the hack in this thread: https://github.com/arendst/Tasmota/discussions/13362

This is contrary to your stated plan of "to adapt to different Wi-Fi networks", as AP mode means that it has not "adapted" to any network.

**Reply by Rqbln** [reply]
Thank you, we're working on the #13362 
Also you were talking about wired flashing for plugs, but we're doing it wireless. Do you know a component we need to buy to flash the plug wired from pc ?

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Getting-Started/
